File: LeetCode/204. Count Primes/Solution.java
public class Solution {//Sieve of Eratosthenes
    public int countPrimes(int n) {
        boolean[] notPrime = new boolean[n];
        if (n < 2) {
            return 0;
        }
        notPrime[0] = notPrime[1] = true;
        int limit = (int) Math.sqrt(n);
        for (int i = 2; i <= limit; i++) {
            if (!notPrime[i]) {
                for (int j = i * i; j < n; j += i) {
                    notPrime[j] = true;
                }
            }
        }
        int count = 0;
        for (boolean b : notPrime) {
            if (!b) {
                count++;
            }
        }
        return count;
    }
}

File: LeetCode/204. Count Primes/solution.py
class Solution(object):
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n < 2: return 0
        ans = [True] * n
        ans[0] = ans[1] = False
        for i in range(2, int(n ** 0.5) + 1):
            if ans[i]:
                for j in range(i * i, n, i):
                    ans[j] = False
        return sum(ans)
